packages/http/src/models/__tests__/Challenge.spec.ts

```diff
@@ -130,6 +134,18 @@ describe('Challenge Model', () => {
     expect(challenge.path[0].result).toBe(PathResult.FAILURE)
   })
 
+  it('has invalid reason when witness is not valid', () => {
+    const challenge = new Challenge(challengeJson([
+      {
+        witnesses: [mockWitness(false)] as HTTPWitnessesObject[],
+        receipt: mockReceipt,
+        geocode: mockGeocode,
+        ...mockPathData,
+      } as HTTPPathObject,
+    ] as HTTPPathObject[]))
+    expect(challenge.path[0].witnesses[0].invalidReason).not.toBeUndefined();
```

Comment:
might want to test that `invalidReason` equals `fake_invalid_reason`

Reply:
good call!